src/domain/substitution.ts

```diff
@@ -0,0 +1,30 @@
+export type SubstitutableVar = 'OWNER' | 'REPO' | 'TAG' | 'VERSION';
+
+export const ALL_SUBSTITUTABLE_VARS: SubstitutableVar[] = [
+  'OWNER',
```

Comment:
my TS isn't strong enough to know whether this list could be shared with the type definition.

Reply:
I think there's a way to do this with an enum.

Reply:
Updated to use an enum and `Object.values()` to get all values.